test/operators/test_block_diag_linear_operator.py

```diff
@@ -67,6 +67,36 @@ def evaluate_linear_op(self, linear_op):
                     actual[i, k, j * 4 : (j + 1) * 4, j * 4 : (j + 1) * 4] = blocks[i, k, j]
         return actual
 
+class TestBlockDiagLinearOperatorMetaClass(unittest.TestCase):
+    def test_metaclass_constructor(self):
+        k, n = 3, 5 # number of blocks, block size
+        b1, b2 = 2, 3 # batch dimensions
+        base_operators = [torch.randn(k, n), torch.randn(b1, b2, k, n)]
+
+        for base_op in base_operators: # repeats tests for both batched and non-batched tensors
+            base_diag = DiagLinearOperator(base_op)
+            linear_op = BlockDiagLinearOperator(base_diag)
+
+            # checks that metaclass constructor returns a diagonal operator
+            # if a DiagLinearOperator is passed to BlockDiagLinearOperator
+            self.assertEqual(type(linear_op), DiagLinearOperator)
+
+            # matrix-vector-multiplication test
+            diag_values = base_op.flatten(start_dim = -2) # operator of non-blocked diagonal values
+            x = torch.randn(diag_values.shape)
```

Comment:
Suggest using this when possible to ensure dtype and device is inherited from arg
```suggestion
            x = torch.randn_like(diag_values)
```